locations/spiders/kkm_vending_krk_pl.py

```diff
@@ -0,0 +1,50 @@
+import re
+from typing import Any
+
+import chompjs
+from scrapy import Spider
+from scrapy.http import Response
+
+from locations.categories import Categories, apply_category
+from locations.hours import DAYS_PL, OpeningHours
+from locations.items import Feature
+
+
+class KrakowPublicTransportVendingMachines(Spider):
+    name = "kkm_vending_krk_pl"
+    item_attributes = {"brand": "KKM", "brand_wikidata": "Q57515549"}
+    start_urls = ["https://kkm.krakow.pl/pl/punkty-sprzedazy-biletow/"]
+
+    def parse(self, response: Response, **kwargs: Any) -> Any:
+        for location_raw in re.findall(r"items\.push\(\s+({.+?})\)", response.text, re.DOTALL):
+            location = chompjs.parse_js_object(re.sub(r"\s:", ":", location_raw))
+            item = Feature()
+            item["ref"] = location["Id"]
+            item["lat"] = location["Latitude"]
+            item["lon"] = location["Longitude"]
+            item["extras"]["location_description"] = location["Location"]
+            if location["OpeningHours"] == "Całą dobę":
+                item["extras"]["opening_hours"] = "24/7"
+            else:
+                entry = OpeningHours()
+                entry.add_ranges_from_string(location["OpeningHours"], DAYS_PL)
+                item["extras"]["opening_hours"] = entry.as_opening_hours()
```

Comment:
```suggestion
                item["opening_hours"] = "24/7"
            else:
                item["opening_hours"] = OpeningHours()
                item["opening_hours"].add_ranges_from_string(location["OpeningHours"], DAYS_PL)
```

`opening_hours` isn't an extra, it's a direct property. Also if we output the `OpeningHours` object, we skip validation if it's the string (`as_opening_hours`) we do some regex on it.